docs/architecture/metadata/structuring-yaml.md

```diff
@@ -0,0 +1,171 @@
+## Creating YAML Files
+
+Metadata YAML files are typically stored within a garden step as `my_dataset.meta.yml`. The conventional structure is as follows:
+
+```yaml
+dataset:
+  update_period_days: ...
+
+tables:
+  my_table:
+    variables:
+      my_var_1:
+        title: ...
+```
+
+To generate a metadata YAML file with pre-populated variable names for an existing garden dataset, execute:
+
+```
+poetry run etl-metadata-export data/garden/my_namespace/my_version/my_dataset -o etl/steps/data/garden/my_namespace/my_version/my_dataset.meta.yml
+```
+
+## Handling Multi-line Strings and Whitespace
+
+Multi-line strings are often sources of confusion. YAML supports two primary styles for writing them:
+
+```yaml
+my_var_1:
+  description: |-
+    The first line
+    and the second
+
+    Third line after line break
+```
+
+```yaml
+my_var_1:
+  description: >-
+    Just a
+    single line
+
+    Second line
+
+
+    Third line after line break
+```
+
+It's up to you whether you want to fit the entire text on the single line and use `>-` or use `|-` and break the lines.
+
+Using `-` after `|` or `>` removes whitespaces at the beginning and end of the string. This is almost always what you want.
+
+
+## Anchors & aliases
+
+[Anchors and aliases](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/) are a native YAML functionality and they can be used to reduce repetition.
+
+Typically we define a special section called `definitions:` at the very top of the file and then we use aliases to refer to these definitions.
+
+An example that reuses `attribution` and `description_key`
+
+```yaml
+definitions:
+  attribution: &attribution Fishcount (2018)
+  description_key_common: &description_key_common
+    - First line.
+    - Second line.
+  description_key_individual:
+    - &third_line Third line.
+
+tables:
+  my_table:
+    variables:
+      my_var_1:
+        description_key: *description_key_common
+        presentation:
+          attribution: *attribution
+      my_var_2:
+        description_key:
+          - *description_key_common
+          - *third_line
+```
+
+
+## Common fields for all indicators
+
+!!! warning "In progress."
+
+To avoid repetition for all indicators, you can use a special section called `common:` under `definitions:`. This section is then merged with sections of all indicators. Using this saves you from repeating the same aliases in indicators.
+
+```yaml
+definitions:
+  common:
+    description_key:
+      - First line.
+      - Second line.
+    presentation:
+      grapher_config:
+        selectedEntityNames:
+          - Germany
+          - Italy
+        topic_tags:
+          - Energy
+
+tables:
+  my_table:
+    variables:
+      my_var_1:
+        # Final description will be First, Second, Third line
+        description_key:
+          - Third line.
+        presentation:
+          # Final tags will be Internet, Energy
+          topic_tags:
+            - Internet
+```
```

Comment:
This logic makes sense as long as we also have the option to override common values. We could do that maybe using the override operator `<<:`?

Reply:
Mentioned `<<:` there and also updated that metadata is overwritten instead of merging.